src/components/MultipleSelectionDialog/MultipleSelectionDialog.js

```diff
@@ -0,0 +1,125 @@
+/**
+ * Copyright (c) 2023, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+import { Dialog, DialogContent, Divider } from '@mui/material';
+import DialogTitle from '@mui/material/DialogTitle';
+import { FormattedMessage } from 'react-intl';
+import FormControlLabel from '@mui/material/FormControlLabel';
+import Checkbox from '@mui/material/Checkbox';
+import React, { useState } from 'react';
+import DialogActions from '@mui/material/DialogActions';
+import Button from '@mui/material/Button';
+import Grid from '@mui/material/Grid';
+
+const MultipleSelectionDialog = ({
+    options,
+    selectedOptions,
+    open,
+    getOptionLabel,
+    handleClose,
+    handleValidate,
+    titleId,
+}) => {
+    const [selectedIds, setSelectedIds] = useState(selectedOptions ?? []);
+    const handleSelectAll = () => {
+        if (selectedIds.length !== options.length) {
+            setSelectedIds(options);
+        } else {
+            setSelectedIds([]);
+        }
+    };
+    const handleOptionSelection = (option) => {
+        setSelectedIds((oldValues) => {
+            if (oldValues.includes(option)) {
+                return oldValues.filter((o) => o !== option);
+            }
+
+            return [...oldValues, option];
+        });
+    };
+
+    return (
+        <Dialog open={open} fullWidth maxWidth={'lg'}>
+            <DialogTitle>{titleId}</DialogTitle>
+            <DialogContent>
+                <Grid container spacing={2}>
+                    <Grid item xs={12}>
+                        <Divider>
+                            <FormControlLabel
+                                label={
+                                    <FormattedMessage
+                                        id={
+                                            'multiple_selection_dialog/selectAll'
+                                        }
+                                    />
+                                }
+                                control={
+                                    <Checkbox
+                                        checked={
+                                            selectedIds.length ===
+                                            options.length
+                                        }
+                                        indeterminate={
+                                            selectedIds.length !==
+                                                options.length &&
+                                            selectedIds.length !== 0
```

Comment:
I think that is better to do

selectedIds.length && selectedIds.length !== options.length 

(it is unnecessary to test selectedIds.length !== options.length if there is no selectedIds)